### PR TITLE
added if clause and unittest for edge case

### LINKED ISSUE #22: Create internship method does not check for deadline date
It is possible to create an internship with a deadline that is a date before the created date. Should be a check for this edge case and another unit test.

### PR DESCRIPTION
**Pull request by louisbank-edu-thailand**

### CODE REVIEW

File: tests/test_internship.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def test_add_internship(
         "company": "Test Company",
         "position": "Facility manager",
         "website": "www.wirecard.com",
-        "deadline": "2023-01-01",
+        "deadline": "2069-01-01",
```
Comment (mythicalbadger) [inline tests/test_internship.py:26]: nice
